picard/tagger.py

```diff
@@ -399,6 +408,36 @@ def _open_files_finished(self, target=None, result=None, error=None):
             for file in new_files:
                 file.load(partial(self._file_loaded, target=target))
 
+    def add_directory_list(self, dir_list, parent=None):
+        if parent:
+            parent = os.path.normpath(parent)
+        number_of_dirs = len(dir_list)
+        if number_of_dirs > 1:
+            if parent:
+                self.window.set_statusbar_message(
+                    N_("Adding %(count)d directories from '%(directory)s' ..."),
+                    {'directory': parent, 'count': number_of_dirs}
+                )
+            else:
+                self.window.set_statusbar_message(
+                    N_("Adding %(count)d directories ..."),
+                    {'directory': parent, 'count': number_of_dirs}
+                )
+        else:
+            self.window.set_statusbar_message(
+                N_("Adding directory: '%(directory)s' ..."),
+                {'directory': dir_list[0]}
+            )
+
+        list_of_files = []
+        recursively = config.setting['recursively_add_files']
+        for directory in dir_list:
+            if recursively:
```

Comment:
I would get rid of this test in loops, and make use of the already imported `itertools.chain`:

```python
    def _add_directory_method(self):
        if config.setting['recursively_add_files']:
            return self._add_directory_recursive
        return self._add_directory_non_recursive

    def add_directory_list(self, dir_list, parent=None):
        if parent:
            parent = os.path.normpath(parent)
        number_of_dirs = len(dir_list)
        if number_of_dirs > 1:
            if parent:
                self.window.set_statusbar_message(
                    N_("Adding %(count)d directories from '%(directory)s' ..."),
                    {'directory': parent, 'count': number_of_dirs}
                )
            else:
                self.window.set_statusbar_message(
                    N_("Adding %(count)d directories ..."),
                    {'directory': parent, 'count': number_of_dirs}
                )
        else:
            self.window.set_statusbar_message(
                N_("Adding directory: '%(directory)s' ..."),
                {'directory': dir_list[0]}
            )

        add_directory = self._add_directory_method()
        self._add_files(list(chain.from_iterable([add_directory(d) for d in dir_list])))

    def add_directory(self, path):
        add_directory = self._add_directory_method()
        self._add_files(add_directory(path))
```